### PR TITLE
feat: add test for pagination cursor

### PR DESCRIPTION
**Pull request by VebjornG**
[Ticket](https://cognitedata.atlassian.net/browse/AH-3373?atlOrigin=eyJpIjoiYjk0NzFmYWQ2Y2I0NGVlYmIxMjFiMzhhYjRiM2ZkNDkiLCJwIjoiaiJ9)

### CODE REVIEW

File: packages/beta/src/types.ts [hunk 1]
```diff
@@ -184,6 +184,7 @@ export interface AlertSort {
 export interface AlertFilterQuery extends FilterQuery {
   filter?: AlertFilter;
   sort?: AlertSort;
+  cursor?: string;
```
Comment (polomani) [inline packages/beta/src/types.ts]: `extends FilterQuery` already gives a cursor

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts [hunk 2]
```diff
@@ -201,3 +201,27 @@ test('sort alerts', async () => {
...
+    sort: {
+      property: 'createdTime',
+      order: 'desc',
+    },
+    cursor: '',
+  });
+  expect(response.items.length).toBe(1);
+  // test for more than 1000 items
+  const response2 = await client.alerts.list({
+    limit: 1001,
```
Comment (polomani) [inline packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: limits is always per page on all other resources on CDF.
to get more then 1000, user would use `autoPagingToArray()`.
check how it's done in other resources, e.g.
```
const timeseries = await response.autoPagingToArray({ limit: 5000 });
```

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts [hunk 3]
```diff
@@ -201,3 +201,29 @@ test('sort alerts', async () => {
...
+    .list({
+      sort: {
+        property: 'createdTime',
+        order: 'desc',
+      },
+      cursor: response?.nextCursor,
+    })
+    .autoPagingToArray({ limit: 1001 });
+
+  expect((await alerts).length).toBeGreaterThan(0);
```
Comment (saagar-cognite) [inline packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: question(non-blocking): is there way to test the number of alerts to be greater than 1000 ?

**Review by saagar-cognite — DISMISSED** [review]
Just 1 nitpick for better testing.

**Comment by saagar-cognite** [comment]
@VebjornG I think you'll need PR title adhering to sdk release guidelines.

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts (hunk 3 above)
Comment (VebjornG) [reply packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: I suppose i could do this: `expect(alerts.length).toBeGreaterThan(1000);` @saagar-cognite

Comment (saagar-cognite) [reply packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: Yes. That'd make sense.
But you might have to make sure that there are more than 1000 alerts present.. So mabye create 1001 alerts and then delete them after the test would be helpful ? 🤔

Comment (VebjornG) [reply packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: I agree. I do so in my latest commit @saagar-cognite 👍🏻

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts [hunk 4]
```diff
@@ -201,3 +201,44 @@ test('sort alerts', async () => {
   });
   expect(response.items.length).toBeGreaterThan(0);
 });
+
+test('cursor pagination', async () => {
+  const client: CogniteClient = setupLoggedInClient();
```
Comment (saagar-cognite) [inline packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: suggestion(blocking): we already have a client instantiated above at starting of `describe` block.
We can avoid initiating it over here.

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts [hunk 5]
```diff
@@ -201,3 +201,44 @@ test('sort alerts', async () => {
...
+    }))
+  );
+
+  const alerts = client.alerts
+    .list({
+      sort: {
+        property: 'createdTime',
+        order: 'desc',
+      },
+      cursor: response?.nextCursor,
```
Comment (saagar-cognite) [inline packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: question(blocking): should this cursor be used from previous response ?
I don't think we need to pass cursor when we're autoPagingToArray.

File: packages/beta/src/__tests__/api/alertsApi.int.spec.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@
 import { CogniteError } from '@cognite/sdk-core';
 import CogniteClient from '../../cogniteClient';
 import { setupLoggedInClient } from '../testUtils';
+//import { AlertCreate } from '@cognite/sdk-beta/dist';
```
Comment (saagar-cognite) [inline packages/beta/src/__tests__/api/alertsApi.int.spec.ts]: nitpick: remove the unused import comment.

**Review by saagar-cognite — APPROVED** [review]
LGTM!

**Pull request merged**